### PR TITLE
Project update

### PR DESCRIPTION
**Pull request by kylefowler**
Upgrade the tooling around the library.

Convert the sample to kotlin.

Add a webview fallback in app.

### CODE REVIEW

File: foursquare-oauth-library/src/main/java/com/foursquare/android/nativeoauth/FoursquareOAuthWebviewActivity.java [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+                        }
+                    }
+
+                    setResult(Activity.RESULT_OK, result);
+                    finish();
+                    return true;
+                }
+                return false;
+            }
+        });
```
Comment (jagl16) [inline foursquare-oauth-library/src/main/java/com/foursquare/android/nativeoauth/FoursquareOAuthWebviewActivity.java]: Can we extract this?

File: foursquare-oauth-library/src/main/java/com/foursquare/android/nativeoauth/FoursquareOAuthWebviewActivity.java [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+        data.putExtra(FoursquareOAuth.INTENT_RESULT_ERROR_MESSAGE, getString(R.string.unsupported_version));
+        setResult(Activity.RESULT_OK, data);
+        finish();
+    }
+
+    /**
+     * Called when the Activity is visible to the user and actively running.
+     * Resumes the WebView.
+     */
+    @SuppressLint("NewApi")
```
Comment (jagl16) [inline foursquare-oauth-library/src/main/java/com/foursquare/android/nativeoauth/FoursquareOAuthWebviewActivity.java]: Which class has this issue, and how it the API not existing being handled?
Is it `ensureCookieSyncManagerAvailable()`?

Comment (kylefowler) [reply foursquare-oauth-library/src/main/java/com/foursquare/android/nativeoauth/FoursquareOAuthWebviewActivity.java]: not sure. this is copied from foursquare-android

File: foursquare-oauth-library/src/main/java/com/foursquare/android/nativeoauth/FoursquareOAuthWebviewActivity.java (hunk 1 above)
Comment (kylefowler) [reply foursquare-oauth-library/src/main/java/com/foursquare/android/nativeoauth/FoursquareOAuthWebviewActivity.java]: technically... yes. With what intent?

Comment (jagl16) [reply foursquare-oauth-library/src/main/java/com/foursquare/android/nativeoauth/FoursquareOAuthWebviewActivity.java]: Simply keeps it easier to read vs having indented code :)

Comment (kylefowler) [reply foursquare-oauth-library/src/main/java/com/foursquare/android/nativeoauth/FoursquareOAuthWebviewActivity.java]: all code that isnt import statements are indented :p

Comment (jagl16) [reply foursquare-oauth-library/src/main/java/com/foursquare/android/nativeoauth/FoursquareOAuthWebviewActivity.java]: 😂😂😂

**Pull request merged**